### PR TITLE
ENH: add error message for merge with Series GH12081

### LINKED ISSUE #12081: DataFrame.merge with Series should give nice error message
Right now trying this results in "IndexError: list index out of range". It should say can't merge DataFrame with a Series...

I know this for quite a while now, but still get trapped on it every once in a while. This would be very helpful for beginners.

Other people also get confused: http://stackoverflow.com/questions/27281734/pandas-merge-on-index-not-working

### PR DESCRIPTION
**Pull request by dan-tee**
closes #12081

### CODE REVIEW

File: pandas/tools/tests/test_merge.py [hunk 1]
```diff
@@ -261,6 +261,20 @@ def test_join_on_fails_with_different_column_counts(self):
                             index=tm.makeCustomIndex(10, 2))
             merge(df, df2, right_on='a', left_on=['a', 'b'])
 
+    def test_join_on_fails_with_left_series(self):
+        with tm.assertRaisesRegexp(ValueError, 'Series'):
+            series = Series([0, 1, 2])
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: iterate thru a bunch of things (e.g. ndarray, number, string) and assert that they raise as well

File: pandas/tools/tests/test_merge.py [hunk 2]
```diff
@@ -261,6 +261,20 @@ def test_join_on_fails_with_different_column_counts(self):
                             index=tm.makeCustomIndex(10, 2))
             merge(df, df2, right_on='a', left_on=['a', 'b'])
 
+    def test_join_on_fails_with_left_series(self):
+        with tm.assertRaisesRegexp(ValueError, 'Series'):
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: add the issue number here as a comment

**Comment by dan-tee** [comment]
@jreback incorporated your comments

File: pandas/tools/tests/test_merge.py [hunk 3]
```diff
@@ -261,6 +261,52 @@ def test_join_on_fails_with_different_column_counts(self):
                             index=tm.makeCustomIndex(10, 2))
             merge(df, df2, right_on='a', left_on=['a', 'b'])
 
+
+    def test_join_on_fails_with_left_series(self):
+        # GH12081
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: you can put all of these test cases in a single test with a loop

**Comment by dan-tee** [comment]
@jreback true, looks much nicer now.

**Comment by jreback** [comment]
looks good. can you add a whatsnew note in API changes.

**Comment by dan-tee** [comment]
@jreback I added the note to whatsnew. I didn't manage to run Sphinx to test build the doc though. Can you give it a spin?

I'll have to clean my development environment and move from using pip to conda. Just don't really have the time now.

**Comment by jreback** [comment]
@dan-tee no looks good. thanks for the PR!

**Pull request closed without merging**